gcloud-java-compute/src/test/java/com/google/gcloud/compute/AddressIdTest.java

```diff
@@ -0,0 +1,114 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class AddressIdTest {
+
+  private static final String PROJECT = "project";
+  private static final String REGION = "region";
+  private static final String NAME = "addr";
+  private static final String GLOBAL_URL =
+      "https://www.googleapis.com/compute/v1/projects/project/global/addresses/addr";
+  private static final String REGION_URL =
+      "https://www.googleapis.com/compute/v1/projects/project/regions/region/addresses/addr";
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testOf() {
+    GlobalAddressId addressId = GlobalAddressId.of(PROJECT, NAME);
+    assertEquals(PROJECT, addressId.project());
+    assertEquals(NAME, addressId.address());
+    assertEquals(GLOBAL_URL, addressId.selfLink());
+    addressId = GlobalAddressId.of(NAME);
+    assertNull(addressId.project());
+    assertEquals(NAME, addressId.address());
+    RegionAddressId regionAddressId = RegionAddressId.of(PROJECT, REGION, NAME);
+    assertEquals(PROJECT, regionAddressId.project());
+    assertEquals(REGION, regionAddressId.region());
+    assertEquals(NAME, regionAddressId.address());
+    assertEquals(REGION_URL, regionAddressId.selfLink());
+    regionAddressId = RegionAddressId.of(RegionId.of(PROJECT, REGION), NAME);
+    assertEquals(PROJECT, regionAddressId.project());
+    assertEquals(REGION, regionAddressId.region());
+    assertEquals(NAME, regionAddressId.address());
+    assertEquals(REGION_URL, regionAddressId.selfLink());
+    regionAddressId = RegionAddressId.of(REGION, NAME);
+    assertNull(regionAddressId.project());
+    assertEquals(REGION, regionAddressId.region());
+    assertEquals(NAME, regionAddressId.address());
+  }
+
+  @Test
+  public void testToAndFromUrl() {
+    GlobalAddressId addressId = GlobalAddressId.of(PROJECT, NAME);
+    compareAddressId(addressId, GlobalAddressId.fromUrl(addressId.selfLink()));
+    thrown.expect(IllegalArgumentException.class);
+    thrown.expectMessage("notMatchingUrl is not a valid global address URL");
```

Comment:
Split this test into two, since exception ends this test and the stuff below the exception doesn't get run.